### PR TITLE
urldecode static filenames

### PR DESCRIPTION
**Pull request by caleb**
If a filename is urlencoded because of spaces and other pesky characters, we need to decode it before trying to find the file on disk.

I'm new to elixir and dynamo, but I ran into this and thought I'd take a stab at fixing it. I added an accompanying regression test as well.

### CODE REVIEW

**Comment by jni-** [comment]
I'm just a beginner too, but I think you could shrink that down with list comprehension and skip the private methods altogether. 

Untested, but something along the lines of : 
`lc char in segment, do: :cowboy_http.urldecode(char)`

Not sure that works, just my 2¢ (although here in Canada we have eliminated pennies, so that would round up to 0$ ;-) )

**Comment by caleb** [comment]
@jni- We could do that, but it would have to look like:

```
lc segment inlist List.wrap(segments), do: :cowboy_http.urldecode(segment)
```

because segments could be a non-list value.

File: lib/dynamo/filters/static.ex [hunk 1]
```diff
@@ -50,6 +50,22 @@ defmodule Dynamo.Filters.Static do
...
+  defp urldecode([]) do
+    []
+  end
+
+  defp urldecode(nil) do
+    nil
+  end
+
+  defp urldecode(segment) when is_binary(segment) do
+    :cowboy_http.urldecode(segment)
```
Comment (josevalim) [inline lib/dynamo/filters/static.ex]: Maybe we should use `URI.decode` instead? :)

**Comment by josevalim** [comment]
Thanks @caleb! And :+1: for something like `lc segment inlist List.wrap(segments), do: URI.decode(segment)`

**Comment by caleb** [comment]
Okay, I've cleaned the patch up.

**Pull request merged**